### ISSUE TITLE
[Shipping Labels] Editing Ship To/From address requires a number in Address 1 field

### ISSUE THREAD

**Issue by joshheald** [woocommerce/woocommerce-ios#5158]
**Describe the bug**
When purchasing a shipping label, the `Address 1` field fails validation if it does not start with a number. The error shown is `House number is missing`.
Many addresses do not start with a number, so this seems like a problematic requirement.

This can be worked around using the `Use Address as Entered` button, however that button is hidden by the keyboard, and it seems as though we should be able to attempt validation of addresses without numbers. In the example video, the address with the number passes validation, even though numbers are (probably) incorrect for the address.

**To Reproduce**
1. Go to the Orders tab
2. Open an order and create a shipping label
3. Tap through to either the `Ship from` or `Ship to` address
4. Enter a new `Address` line 1, which starts with a letter (it can have numbers anywhere else)
5. Tap `Done`
6. The error will be displayed

**Screenshots**

https://user-images.githubusercontent.com/2472348/136408898-fa19e286-cda3-4f26-ae87-8c38f1974e40.mp4

**Expected behavior**
Addresses without numbers should be accepted for validation attempts.

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Mobile Environment**
Please include:
- Device: iPhone 7 Plus
- iOS version: 14.8
- WooCommerce iOS version: 7.7

**Comment by itsmeichigo** [comment]
@joshheald - Thanks for reporting your concern! 

We're not currently doing local validation for address 1 (other than it has to be not empty) - so this error certainly comes from remote validation API. In cases where user really wants to ignore the validation issue, they can select the button "Use Address as Entered" at the bottom of the screen. 

I know this button is hidden by the keyboard when editing the address, so it can be annoying if user wants to skip validation. It would be great if we make it possible to dismiss the keyboard, but I believe it makes sense to make address validation mandatory.

**Comment by joshheald** [comment]
@itsmeichigo Ah sorry, it felt local to me! 

What do we use for the remote validation? It seems like quite a naïve approach they're taking at the moment... there are a lot of addresses which don't start with a number, and it's quite frustrating when forms don't accept your address (speaking from experience, as someone who has a `/` in his flat number!)

A merchant may also feel concerned about it: their customer has entered an address, if it doesn't have a house number and they add one, are they making it less likely that the order will be successfully delivered? But if they don't add a house number, and use an unvalidated address (if they see that hidden button) are they then making it more likely for the delivery to fail that way too?

I think it would be preferable to find a way for a merchant in that situation to get the full response from the remote API (I'm assuming these are the same APIs!), which gives a suggested updated address from their database, and allows the merchant to choose between them. This message seems like field input validation from the API which it would be quite nice to turn off, and just let it try to figure out the address suggestions without a house number if there wasn't one provided by the customer/merchant. 

I wonder whether we can configure the request or API set up differently to avoid this one?

**Comment by joshbetz** [comment]
@joshheald I copied you on a thread to discuss fixing this in the API https://wp.me/p91TBi-6zK

**Comment by joshheald** [comment]
Automated triage note: this was tested and commented by Codex using device/simulator automation.

Attempted on iPad Pro 11-inch (M4) simulator running iOS 18.6. I could not fully reproduce the live label-purchase validation path because WDA was unavailable for navigation and this requires a shipping-label-capable order plus remote address validation. Current code still handles remote "House number is missing" responses; destination addresses can continue via "Use address as entered", while origin addresses remain blocked on that validation failure. So this still looks relevant, but I did not get a live UI reproduction in this pass.

I am leaving this GitHub issue open and applying the `triage: automation attempted` label for follow-up visibility.

**Comment by dangermattic** [comment]
Thanks for reporting! 👍

**Comment by JorgeMucientes** [comment]
Closing this because the mobile shipping labels flow was fully revamped in 2025, and this issue predates that revamp / targets legacy shipping labels behavior. If this still reproduces in the current shipping labels flow, please open a new issue with updated steps and screenshots.
